packages/component/src/BasicTranscript.js

```diff
@@ -30,83 +28,84 @@ const FILLER_CSS = css({
 });
 
 const LIST_CSS = css({
-  listStyleType: 'none',
-
-  '& > li.hide-timestamp .transcript-timestamp': {
-    display: 'none'
-  }
+  listStyleType: 'none'
 });
 
-const DEFAULT_GROUP_TIMESTAMP = 300000; // 5 minutes
+function useMemoize(fn) {
+  return useMemo(() => {
+    let cache = [];
+
+    return run => {
+      const nextCache = [];
+      const result = run((...args) => {
+        const { result } = cache.find(
+          ({ args: cachedArgs }) =>
+            args.length === cachedArgs.length && args.every((arg, index) => Object.is(arg, cachedArgs[index]))
+        ) || { result: fn(...args) };
```

Comment:
Could you explain this one to me?

Reply:
Memoize function usually cache a single result.

This `useMemoize` hook will cache all the result within the same run.

For example,

```js
const expensiveSum = (x, y) => x + y;

const memoizedSum = useMemoize(expensiveSum);

// Start a run, all calls to sum() will be cached.
memoizedSum(sum => {
  sum(1, 2); // Not cached, return 3.
  sum(1, 2); // Cached, return 3.
  sum(2, 4); // Not cached, return 6.
  sum(1, 2); // Cached, return 3. This is cached because it is inside the same run.
});

// After the run, 1 + 2 = 3, and 2 + 4 = 6 is cached.

// Start another run with previous cache
memoizedSum(sum => {
  sum(1, 2); // Cached, return 3.
  sum(3, 6); // Not cached, return 9.
});

// After the run, 1 + 2 = 3 and 3 + 6 = 9 is cached. 2 + 4 is dropped.
```

Reply:
I just added test, found a flaw, fixed. 😆